### PR TITLE
Implement semantic-gitcommit versioning as Github Workflow

### PR DESCRIPTION
**Pull request by mikenye**
Hi Team,

As-per #29, this PR implements semantic-gitcommit versioning using the action [paulhatch/semantic-version](https://github.com/marketplace/actions/git-semantic-version) for the **Deploy** workflow.

> The commit messages for the span of commits from the last tag are checked for the presence of the designated terms (`(MAJOR)` or `(MINOR)` by default), if a term is encountered that commit is treated as the start of a major or minor version instead of the default patch level.

The version string is formatted as follows: `${major}.${minor}.${patch}.${increment}-${short_commit_hash}`.

The version string is passed as a docker build argument, and added into the `go build` command to set the `main.version` variable for the binaries in the containers.

I've updated the `pw_discord_bot`, `pw_ingest`, `pw_router` and `pw_ws_broker`'s `main.go` file to include the version, which is applied during build.

Docker images will be tagged as:

- `ghcr.io/plane-watch/pw-pipeline:${binary}` (this is the functional equivalent of `latest`)
- `ghcr.io/plane-watch/pw-pipeline:${binary}_${version}`

**Images will be pushed to ghcr.io on push to the main branch.**

If the workflow is run manually, images will be built only. We can adjust this behaviour in future if required.

The action also gives us the ability to create a github release by writing a link to its logs, eg: https://github.com/plane-watch/pw-pipeline/runs/6125969323?check_suite_focus=true#step:4:16

---

I also implemented `go test -race ...`, which is why this workflow is failing. Race detector is finding race conditions.

See: <https://github.com/plane-watch/pw-pipeline/runs/6126373764?check_suite_focus=true>

### CODE REVIEW

**Comment by mikenye** [comment]
@bluntelk I also linted the YAML files in case you're wondering why they're changed.

File: .github/workflows/cancel_dupes.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+---
```
Comment (bluntelk) [inline .github/workflows/cancel_dupes.yml:1]: do you need this set of dashes?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+---
```
Comment (bluntelk) [inline .github/workflows/codeql-analysis.yml:1]: more dashes

**Review by bluntelk — APPROVED** [review]
Just a question about some extra -'s

File: .github/workflows/cancel_dupes.yml (hunk 1 above)
Comment (mikenye) [reply .github/workflows/cancel_dupes.yml:1]: yamllint complains without them. Apparently starting a YAML file with three dashes is "start of document".

Ref: https://stackoverflow.com/questions/50788277/why-3-dashes-hyphen-in-yaml-file

Comment (mikenye) [reply .github/workflows/cancel_dupes.yml:1]: Here's the yamllint rule: https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.document_start

**Pull request merged**